(feat) follow up changes on the case management: discontinuity of case

its-kios09 opened the pull request:
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).


## Summary
With reference on this closed PR: https://github.com/palladiumkenya/kenyaemr-esm-3.x/pull/520/commits/29e5d488bebe824f4aa8882a865449f1b7011bf9#
<!--
Required.
Please describe what problems your PR addresses.
-->


## Screenshots
![caseUI](https://github.com/user-attachments/assets/1c24fc48-2814-4c6c-8cf3-ff0c4a5b433d)

*None.*
<!--
Optional.
If possible, please insert any screenshots/videos of your changes here.
Don't forget to remove the *None.* above if you do fill this section.
-->


## Related Issue

*None.*
<!--
Required if applicable.
If present, please link any related issue here, e.g. "https://issues.openmrs.org/browse/123").
Don't forget to remove the *None.* above if you do fill this section.
-->


## Other

*None.*
<!--
Optional.
Anything else that isn't covered by one of the sections above.
Don't forget to remove the *None.* above if you do fill this section.
-->

ojwanganto commented:
@donaldkibet @makombe @Omoshlawi please help with reviews.

ojwanganto commented on packages/esm-billing-app/translations/en.json:
```diff
@@ -72,10 +70,7 @@
-  "clockingIn": "Clocking in",
-  "clockingOut": "Clocking Out",
-  "clockOut": "Clock Out",
+  "clockInTime": "",
```
Not relevant here

ojwanganto commented on packages/esm-patient-clinical-view-app/translations/en.json:
```diff
@@ -156,9 +159,13 @@
   "releaseBody": "Release Body",
   "reportingMonth": "Month",
   "reportingYear": "Year",
+  "rlshipConfirmationText": "This will end the relationship. Are you sure you want to proceed?",
```
let's just have relationship in full

ojwanganto commented on packages/esm-patient-clinical-view-app/translations/en.json:
```diff
@@ -156,9 +159,13 @@
   "reportingMonth": "Month",
   "reportingYear": "Year",
+  "rlshipConfirmationText": "This will end the relationship. Are you sure you want to proceed?",
+  "RlshipErrorMessage": "Request Failed",
```
same here

ojwanganto commented on packages/esm-patient-clinical-view-app/translations/en.json:
```diff
@@ -156,9 +159,13 @@
   "reportingYear": "Year",
+  "rlshipConfirmationText": "This will end the relationship. Are you sure you want to proceed?",
+  "RlshipErrorMessage": "Request Failed",
+  "savedRlship": "Relationship ended successfully",
```
same here

donaldkibet commented on packages/esm-patient-clinical-view-app/src/case-management/modal/case-management-modal.component.tsx:
```diff
@@ -0,0 +1,106 @@
+      showSnackbar({
+        kind: 'error',
+        title: t('relationshipError', 'Relationship Error'),
+        subtitle: t('RlshipErrorMessage', 'Request Failed'),
```
```suggestion
        subtitle: t('rlshipErrorMessage', 'Request Failed'),
```

donaldkibet commented:
@its-kios09 please fix the failing build

its-kios09 commented on packages/esm-patient-clinical-view-app/src/case-management/metrics/case-management-header.component.tsx:
```diff
@@ -14,14 +18,18 @@ const MetricsHeader = () => {
     });
   };
+
+  const isDiscontinuationTab = activeTabIndex === 1;
```
@donaldkibet The tab index corresponds to the order of the tabs as defined in the TabList: Index 0: The first tab, "Active Cases." Index 1: The second tab, "Discontinuation Cases." This order is fixed and consistent throughout the component. Using these indices ensures that the correct content and functionality (e.g., disabling the "Add Case" button in the "Discontinuation Cases" tab) is tied to the appropriate tab.

its-kios09 commented on packages/esm-patient-clinical-view-app/src/case-management/workspace/case-management.scss:
```diff
@@ -19,8 +19,8 @@
-.datePickerInput {
-  width: 23rem;
+.formDatePicker input {
+  min-width: 24rem;
```
![screenshot](https://github.com/user-attachments/assets/51d182f1-7893-4914-8be0-1d9c28f1b9d3)

its-kios09 replied:
![ser](https://github.com/user-attachments/assets/8f699307-8bd6-4286-9a90-83708db01472)

its-kios09 commented:
> @its-kios09 please fix the failing build

@donaldkibet addressed the build failure

donaldkibet commented on packages/esm-patient-clinical-view-app/src/case-management/table/case-management-list-inactive-component.tsx:
```diff
@@ -52,21 +53,24 @@ const CaseManagementListInActive: React.FC<CaseManagementListInActiveProps> = ({
-      dateofstart: new Date(caseData.startDate).toLocaleDateString(),
-      dateofend: new Date(caseData.endDate).toLocaleDateString(),
-    }));
+  const tableRows = filteredCases?.slice((currentPage - 1) * pageSize, currentPage * pageSize).map((caseData) => ({
```
why are we modifying the filteredCases, we already have the pagination already done

donaldkibet commented on packages/esm-patient-clinical-view-app/src/case-management/modal/case-management-modal.component.tsx:
```diff
@@ -0,0 +1,106 @@
+import React from 'react';
```
IMO we should have this as a workspace item since its a form

its-kios09 commented on packages/esm-billing-app/translations/en.json:
```diff
@@ -72,10 +70,7 @@
-  "clockingIn": "Clocking in",
-  "clockingOut": "Clocking Out",
-  "clockOut": "Clock Out",
+  "clockInTime": "",
```
I have fixed to avoid empty translation

its-kios09 commented:
ping @donaldkibet I have addressed all the requested changes for the clockin having a empty translation I have addressed it here on this PR
![workspace](https://github.com/user-attachments/assets/ad27196c-4efa-4a14-b7b0-339b24772102)

donaldkibet commented:
> ping @donaldkibet I have addressed all the requested changes for the clockin having a empty translation I have addressed it here on this PR ![workspace](https://private-user-images.githubusercontent.com/67967749/402541783-ad27196c-4efa-4a14-b7b0-339b24772102.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY3ODI3MDIsIm5iZiI6MTczNjc4MjQwMiwicGF0aCI6Ii82Nzk2Nzc0OS80MDI1NDE3ODMtYWQyNzE5NmMtNGVmYS00YTE0LWI3YjAtMzM5YjI0NzcyMTAyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTEzVDE1MzMyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyZjA5MzIxZTNkNzc4MzM0NWY5MWI2MGNjMTU0ZDJkMzIyZDQyYzNhZTkwOTFiNDY3MTgzNDFmZDQyYWNhNzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Y6LR6cN8-6wrP0QnggqzEvaBU6gEgpoKCCOzWg6O4FQ) [ ![workspace](https://private-user-images.githubusercontent.com/67967749/402541783-ad27196c-4efa-4a14-b7b0-339b24772102.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY3ODI3MDIsIm5iZiI6MTczNjc4MjQwMiwicGF0aCI6Ii82Nzk2Nzc0OS80MDI1NDE3ODMtYWQyNzE5NmMtNGVmYS00YTE0LWI3YjAtMzM5YjI0NzcyMTAyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTEzVDE1MzMyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyZjA5MzIxZTNkNzc4MzM0NWY5MWI2MGNjMTU0ZDJkMzIyZDQyYzNhZTkwOTFiNDY3MTgzNDFmZDQyYWNhNzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Y6LR6cN8-6wrP0QnggqzEvaBU6gEgpoKCCOzWg6O4FQ) ](https://private-user-images.githubusercontent.com/67967749/402541783-ad27196c-4efa-4a14-b7b0-339b24772102.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY3ODI3MDIsIm5iZiI6MTczNjc4MjQwMiwicGF0aCI6Ii82Nzk2Nzc0OS80MDI1NDE3ODMtYWQyNzE5NmMtNGVmYS00YTE0LWI3YjAtMzM5YjI0NzcyMTAyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTEzVDE1MzMyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyZjA5MzIxZTNkNzc4MzM0NWY5MWI2MGNjMTU0ZDJkMzIyZDQyYzNhZTkwOTFiNDY3MTgzNDFmZDQyYWNhNzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Y6LR6cN8-6wrP0QnggqzEvaBU6gEgpoKCCOzWg6O4FQ) [ ](https://private-user-images.githubusercontent.com/67967749/402541783-ad27196c-4efa-4a14-b7b0-339b24772102.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY3ODI3MDIsIm5iZiI6MTczNjc4MjQwMiwicGF0aCI6Ii82Nzk2Nzc0OS80MDI1NDE3ODMtYWQyNzE5NmMtNGVmYS00YTE0LWI3YjAtMzM5YjI0NzcyMTAyLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTEzVDE1MzMyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyZjA5MzIxZTNkNzc4MzM0NWY5MWI2MGNjMTU0ZDJkMzIyZDQyYzNhZTkwOTFiNDY3MTgzNDFmZDQyYWNhNzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Y6LR6cN8-6wrP0QnggqzEvaBU6gEgpoKCCOzWg6O4FQ)

We will need to work on the `Discard` and `Save` button to be consistent with the rest of the App, e.g

Conditions workspace

![Screenshot 2025-01-13 at 18 35 17](https://github.com/user-attachments/assets/4f100c31-43a8-4d58-8167-19fef8a18ff3)
